runtests.py

```diff
@@ -1219,6 +1243,10 @@ def run_distutils(self, test_directory, module, workdir, incdir,
             if self.language == 'cpp':
                 # Set the language now as the fixer might need it
                 extension.language = 'c++'
+                if self.extra_directives and self.extra_directives.get('cpp_locals', None):
```

Comment:
By making sure that we always have a dict, this can be simplified.
```suggestion
                if self.extra_directives.get('cpp_locals'):
```